src/_guides/libraries/java-interop.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Java interop using package:jnigen"
+description: "To use Java in your Dart program, use package:jnigen."
+jnigen: "https://pub.dev/packages/jnigen"
+jni: "https://pub.dev/packages/jni"
+example: "https://github.com/HosseinYousefi/jnigen_example/tree/main"
+jnidoc: "https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/jniTOC.html"
+---
+
+Dart mobile, command-line, and server apps
+running on the [Dart Native platform](/overview#platform), on Android, Windows
```

Comment:
Is macOS not supported yet?

Reply:
Looks like it's not https://github.com/dart-lang/jnigen#supported-platforms

Reply:
Ah, thanks for the link! Seems `macOS` is supported, just not for Flutter yet according to the table there. Would be nice to include macOS, since many Dart developers use that as their development platform. If it's important that Flutter doesn't support it, we could add a Flutter note or a simple exception in parentheses.